src/main/java/com/sarality/form/binding/CheckBoxGroupBinding.java

```diff
@@ -118,6 +112,17 @@ public void setValueList(List<String> textValueList) {
     }
   }
 
+  @Override public void setValueProvider(ControlValueProvider valueProvider) {
```

Comment:
We clearly need to redesign this, because if we can't go back in the code and understand what is going on, this badly designed code. I'll do that later, but for now we have to live with whatever design I came up with earlier.

The question to ask here, is why does setValueProvider need to be overridden here while it is not overidden anywhere else and 2) Why does the renderer not take care of this in super class itself? Do we need to introduce a CheckBoxGroupRenderer instead?